src/qt/bitcoingui.cpp

```diff
@@ -661,12 +662,12 @@ void BitcoinGUI::trayIconActivated(QSystemTrayIcon::ActivationReason reason)
 
 void BitcoinGUI::optionsClicked()
 {
-    if(!clientModel || !clientModel->getOptionsModel())
-        return;
+    openOptionsDialogWithTabIndex(0);
+}
 
-    OptionsDialog dlg(this, enableWallet);
-    dlg.setModel(clientModel->getOptionsModel());
-    dlg.exec();
+void BitcoinGUI::proxyIconClicked()
+{
+    openOptionsDialogWithTabIndex(2);
```

Comment:
nit: not sure if setting by the object name would make more sense.
Something like `setCurrentWidget(ui->tabNetwork)`.

Reply:
Would need to get `ui->tabNetwork` and `ui->tabMain` at `BitcoinGUI` level and in my opinion that is not worth it. The main concern is related to magic values or if the tabs change. Magic values are not a real concern imo because they are "labeled" by the method calling them (ex: `proxyIconClicked`). I don't think the order of tabs will change. Will it?

I tried something similar to this:

```
     OptionsDialog dlg(this, enableWallet);
-    dlg.setCurrentTabIndex(tabIndex);
+    if (tabIndex == 0) {
+      dlg.setTabMain();
+    } else {
+      dlg.setTabNetwork();
+    }
```

It adds 2 new methods to `OptionsDialog`, but still needed to pass `tabIndex` because I would need to make a methods which return `ui->tabNetwork` and `ui->tabMain`.

So I think I prefer the current implementation. Opinions?

Reply:
Could create an enum for tab indexes? So it would be something like
```cpp
void BitcoinGUI::openOptionsDialogWithTab(OptionsDialog::Tab tab)
```

Reply:
Added the `enum`, makes the code more readable, thx